toolchain/check/import_cpp.cpp

```diff
@@ -485,18 +486,27 @@ auto ImportCppFiles(Context& context,
   return std::move(generated_ast);
 }
 
-// Looks up the given name in the Clang AST in a specific scope. Returns the
-// lookup result if lookup was successful.
-static auto ClangLookupName(Context& context, SemIR::NameScopeId scope_id,
-                            SemIR::NameId name_id)
-    -> std::optional<clang::LookupResult> {
-  std::optional<llvm::StringRef> name =
-      context.names().GetAsStringIfIdentifier(name_id);
-  if (!name) {
-    // Special names never exist in C++ code.
-    return std::nullopt;
+// Returns the Clang `DeclContext` for the given name scope. Return the
+// translation unit decl if not scope is provided.
```

Comment:
```suggestion
// translation unit decl if no scope is provided.
```
Does the "no scope is provided" case happen? That sounds like it would probably be a bug in the caller.

Reply:
When calling `ClangLookupDeclarationName()` from `ImportOperatorFromCpp()` I didn't pass a scope because I couldn't find a scope that makes sense to pass.
We could improve this if you think operators in Carbon could have a C++ scope context.

Reply:
Hm, I see.

Eventually, we probably shouldn't be doing a qualified lookup at all here -- instead, I think we should perform an ADL-only lookup for operators (`Sema::ArgumentDependentLookup`). But I think we'd need the machinery from #5891 to map Carbon types into C++ types before we could do that.

Reply:
But to be clear: let's keep going with this qualified lookup approach in this PR. I don't think there's much here we'll want to roll back when we add support for argument-dependent operator lookup.

Reply:
Thanks for the details!
Added a TODO.